crates/zk-por-core/src/bin/prover.rs

```diff
@@ -106,23 +125,60 @@ fn main() {
             "parse {} times, with number of accounts {}, number of batches {}",
             parse_num,
             account_num,
-            account_num / batch_size
+            expected_batch_num,
         );
 
+        let batch_idx_base = batch_proofs.len();
+        let root_hashes: Vec<HashOut<F>> = accounts
+            .par_chunks(batch_size)
+            .enumerate()
+            .map(|(i, account_batch)| {
+                let batch_idx = batch_idx_base + i;
+                let mst = MerkleSumTree::new_tree_from_accounts(&account_batch.to_vec());
+
+                let global_mst = GLOBAL_MST.get().unwrap();
+                let mut _g = global_mst.write().expect("unable to get a lock");
+
+                for inner_tree_node_idx in 0..batch_size * 2 - 1 {
+                    _g.set_batch_hash(
```

Comment:
[optimization] we can set in a batch fasion (out side the par_chunks). so we only acquire lock once; (my previous code not optimized); pls help optimize this part; simlar to recursive